vcr/verifier/interface.go

```diff
@@ -0,0 +1,42 @@
+package verifier
+
+import (
+	"errors"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+	"github.com/nuts-foundation/nuts-node/vcr/signature/proof"
+	"time"
+)
+
+// Verifier defines the interface for verifying verifiable credentials.
+type Verifier interface {
+	// Verify checks credential on full correctness. It checks:
+	// validity of the signature
+	// if it has been revoked
+	// if the issuer is registered as trusted
+	Verify(credential vc.VerifiableCredential, allowUntrusted bool, checkSignature bool, validAt *time.Time) error
+	// Validate checks the verifiable credential technical correctness
+	Validate(credentialToVerify vc.VerifiableCredential, at *time.Time) error
+	// IsRevoked checks if the credential is revoked
+	IsRevoked(credentialID ssi.URI) (bool, error)
+	// CheckAndStoreRevocation accepts a signed document, checks the signature and type.
+	// If all is valid, it stores the revocation.
+	CheckAndStoreRevocation(document proof.SignedDocument) error
```

Comment:
This function feels a bit misplaced here. Maybe because it combines both verifying the revocation (probably OK) and storing it, a responsibility of the `Store` (below), causing it to just pass it on? Hence the long name.

Possible improvement, 2 options I see (although not fantastic either):

- Split `CheckAndStoreRevocation` in `VerifyRevocation` and `RegisterRevocation`
- Just provide `VerifyRevocation` here and have the caller store it directly in the store